core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceInfo.java

```diff
@@ -69,4 +86,13 @@ String getAdvertisingId() {
     boolean isAdTrackingEnabled() {
         return this.adTrackingEnabled;
     }
+
+    void clearAdvertisingId() {
+        this.advertisingId = null;
+        this.adTrackingEnabled = (this.advertisingId != null);
```

Comment:
Let's simplify this:
```suggestion
        this.adTrackingEnabled = false;
```

Reply:
updated